zebra-consensus/src/error.rs

```diff
@@ -121,7 +121,7 @@ pub enum TransactionError {
         transaction_hash: zebra_chain::transaction::Hash,
     },
 
-    #[error("coinbase transaction failed subsidy validation")]
+    #[error("coinbase transaction failed subsidy validation: {0}")]
```

Comment:
Would the debug output in any of these would include useful details and minimal noise, or the opposite?

```suggestion
    #[error("coinbase transaction failed subsidy validation: {0:?}")]
```

Reply:
Since `#[error()]` is used for the Display impl, I think we should use Display with Display, and Debug with Debug, i.e. leave it as `{0}`

Currently it's, e.g.:

"coinbase transaction failed subsidy validation: funding stream expected output not found"

With `{0:?}` it would be:

"coinbase transaction failed subsidy validation: FundingStreamNotFound"

Where we want/need we can convert the top error to Debug which would give us:

"Subsidy(FundingStreamNotFound)"